Repositories/ProductMongoDBRepository.cs

```diff
@@ -0,0 +1,100 @@
+﻿using MongoDB.Driver;
+using InventoryManagement.Models;
+using System.Collections.Generic;
+using MongoDB.Bson;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using System.IO;
+namespace InventoryManagement.Repositories
+{
+    public class ProductMongoDBRepository : IProductRepository
+    {
+        private IMongoClient _client;
+        private IMongoCollection<BsonDocument> _collection;
+        private IMongoDatabase _database;
+
+
+        public void SetupDB()
+        {
+            var config = new ConfigurationBuilder().SetBasePath(Directory.GetCurrentDirectory()).AddJsonFile("appsettings.json").Build();
+            string connectionStrong = config["MongoDB:connectionString"];
+
+            try
+            {
+                _client = new MongoClient(connectionStrong);
+                _database = _client.GetDatabase("inventory");
+                _collection = _database.GetCollection<BsonDocument>("Inventory");
```

Comment:
You can use strongly typed collection instead, for example
```csharp
_database.GetCollection<ProductDocument>("Inventory")
```
and then you can do something like:
```csharp
var filter = Builders<ProductDocument>.Filter.Eq(d => d.Name, name);
```
instead of relying on string names.